### DISCUSSION TITLE
How to log responses with an interceptor

### DISCUSSION THREAD

**Discussion by tomasol** [hyperium/tonic#752] [Q&A] [answered]
I would like to use a middleware to log response errors. I looked at the [server interceptor example](https://github.com/hyperium/tonic/blob/master/examples/src/interceptor/server.rs) but I can't figure out how to interact with the response.

**Comment by davidpdrsn** [comment, marked as answer]
See the [`Interceptor` docs](https://docs.rs/tonic/0.5.2/tonic/service/trait.Interceptor.html).

TLDR: Use [`TraceLayer::new_for_grpc`](https://docs.rs/tower-http/0.1.1/tower_http/trace/index.html) from tower-http.
